lib/CBonusTypeHandler.cpp

```diff
@@ -95,12 +95,6 @@ std::string CBonusTypeHandler::bonusToGraphics(const std::shared_ptr<Bonus> & bo
 
 	switch(bonus->type)
 	{
-	case Bonus::SECONDARY_SKILL_PREMY:
-		if(bonus->subtype == SecondarySkill::RESISTANCE)
-		{
-			fileName = "E_DWARF.bmp";

```

Comment:
Do dwarfs still get have correct icon for their ability?

Reply:
Will check.

Reply:
With vcmi-extras dwarves has correct bonus icon for magic resistance.